slateratom/lib/confinement.F90

```diff
@@ -0,0 +1,514 @@
+                  & alpha(ii, jj), kk, alpha(ii, ll), mm, ii, vconf_matrixelement)
+
+              vconf_matrix(ii, ss, tt) = vconf_matrixelement
+              vconf_matrix(ii, tt, ss) = vconf_matrixelement
+
+            end do
+          end do
+        end do
+      end do
+    end do
+
+  end subroutine TConf_getSupervec
+
+
+  !> Calculates a single matrix element of the confinement potential.
+  pure subroutine dft_conf_matrixelement(num_mesh_points, weight, abcissa, vconf, alpha1, poly1,&
+      & alpha2, poly2, ll, conf_matrixelement)
+
+    !> number of numerical integration points
+    integer, intent(in) :: num_mesh_points
+
+    !> numerical integration weights
+    real(dp), intent(in) :: weight(:)
+
+    !> numerical integration abcissas
+    real(dp), intent(in) :: abcissa(:)
+
+    !> confinement potential on grid
+    real(dp), intent(in) :: vconf(:)
+
+    !> basis exponent of 1st basis
+    real(dp), intent(in) :: alpha1
+
+    !> highest polynomial order in 1st basis shell
+    integer, intent(in) :: poly1
+
+    !> basis exponent of 2nd basis
+    real(dp), intent(in) :: alpha2
+
+    !> highest polynomial order in 2nd basis shell
+    integer, intent(in) :: poly2
+
+    !> angular momentum
+    integer, intent(in) :: ll
+
+    !> single matrix element of the confinement potential
+    real(dp), intent(out) :: conf_matrixelement
+
+    !> stores product of two basis functions and r^2
+    real(dp) :: basis
+
+    !> auxiliary variable
+    integer :: ii
+
+    conf_matrixelement = 0.0_dp
+
+    do ii = 1, num_mesh_points
+      basis = basis_times_basis_times_r2(alpha1, poly1, alpha2, poly2, ll, abcissa(ii))
+      conf_matrixelement = conf_matrixelement + weight(ii) * vconf(ii) * basis
+    end do
+
+  end subroutine dft_conf_matrixelement
+
+
+  !> Returns the Power potential for a given radial distance.
+  elemental impure function getPowerPotential(rr, r0, power) result(pot)
+
+    !> radial distance
+    real(dp), intent(in) :: rr
+
+    !> confinement radius
+    real(dp), intent(in) :: r0
+
+    !> confinement power
+    real(dp), intent(in) :: power
+
+    !> Power potential at evaluation point
+    real(dp) :: pot
+
+    @:ASSERT(rr >= 0.0_dp)
+
+    pot = (rr / r0)**power
+
+  end function getPowerPotential
+
+
+  !> Returns the Woods-Saxon potential for a given radial distance.
+  !! see J. Chem. Theory Comput. 12, 1, 53-64 (2016) eqn. 4.
+  elemental impure function getWSPotential(rr, ww, aa, r0) result(pot)
+
+    !> radial distance
+    real(dp), intent(in) :: rr
+
+    !> potential height (W)
+    real(dp), intent(in) :: ww
+
+    !> potential slope (a)
+    real(dp), intent(in) :: aa
+
+    !> half-height radius (r0)
+    real(dp), intent(in) :: r0
+
+    !> Woods-Saxon potential at evaluation point
+    real(dp) :: pot
+
+    @:ASSERT(rr >= 0.0_dp)
+
+    pot = ww / (1.0 + exp(-aa * (rr - r0)))
+
+  end function getWSPotential
+
+
+  ! !> Calculates analytic matrix elements of confining potential.
```

Comment:
Any reason to keep this routine?

Reply:
I just wanted a version that is compatible with the new structure of the module to appear in the history. Removed in the added commit.